src/main.c

```diff
@@ -134,6 +130,13 @@ int main(void)
 
   TU_LOG2("DFU init...\r\n");
   board_dfu_init();
+
+// Some platforms initialize flash earlier to check for a valid image
+#ifndef EARLY_FLASH_INIT  

```

Comment:
we should completely remove the board_flash_init() and leave it up to `board_init()` or `board_dfu_init()` to does its work.